graphql_compiler/compiler/ir_lowering_gremlin.py

```diff
@@ -252,6 +254,31 @@ def to_gremlin(self):
         return u'findAll{{entry -> {}}}'.format(self.predicate.to_gremlin())
 
 
+class GremlinFoldedTraverse(Traverse):
+    """A Gremlin-specific Traverse block to be used only within @fold scopes."""
+
+    def __init__(self, traverse_block):
```

Comment:
This would perhaps be better suited to a classmethod factory function `from_traverse` or something like that, otherwise it's a confusing pattern -- the constructor of a subclass of `Traverse` requires a `Traverse` object to initialize itself.